compiler/src/iree/compiler/Dialect/Flow/IR/FlowOps.td

```diff
@@ -24,6 +24,27 @@ class FLOW_PureOp<string mnemonic, list<Trait> traits = []> :
 // Partitioned regions
 //===----------------------------------------------------------------------===//
 
+def FLOW_DispatchRegionOp : FLOW_PureOp<"dispatch.region", [
+    SingleBlockImplicitTerminator<"IREE::Flow::ReturnOp">]> {
```

Comment:
Does this not need to implement the `Util_ShapeAwareOp` interface like `DispatchWorkgroupsOp` below?

Reply:
Yes it does, see #9963. (This PR depends on #9963 but I haven't updated the first commit yet.)